ktor-server/ktor-server-host-common/jvmAndNix/src/io/ktor/server/engine/BaseApplicationResponse.kt

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2014-2021 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.server.engine
+
+import io.ktor.http.*
+import io.ktor.http.content.*
+import io.ktor.server.application.*
+import io.ktor.server.response.*
+import io.ktor.util.*
+import io.ktor.utils.io.*
+import kotlinx.coroutines.*
+
+/**
+ * Base class for implementing an [ApplicationResponse]
+ */
+public expect abstract class BaseApplicationResponse(call: ApplicationCall) : ApplicationResponse {
```

Comment:
Can you explain why expect/actual is needed here?

Reply:
to minimize `shared` effects to the jvm side